### ISSUE TITLE
CI: build-and-test / build-package (rpm) fails due to CentOS 8 EOL

### ISSUE THREAD

**Issue by djaglowski** [open-telemetry/opentelemetry-collector-contrib#7529]
CentOS 8 has reached EOL and [has been made unavailable at the previously used url](https://forums.centos.org/viewtopic.php?f=54&t=78708#p330689). This is causing failures when creating the RPM package.

```
Step 3/6 : RUN dnf install -y initscripts
 ---> Running in 27f19211d0d1
CentOS Linux 8 - AppStream                      128  B/s |  38  B     00:00    
Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
```

I'm not very familiar with our release process and am unsure the overall implications & requirements for switching distributions, but I opened #7526 to test whether switching to `rockylinux` (apparently one of the successors to CentOS) will pass CI. It does, but I expect there is more to this decision and probably at least another code reference that needs to be updated for the actual release. Thoughts?

**Comment by djaglowski** [comment]
Discussed in SIG - the usage of centos 8 is really a general test meant to represent whether RPMs can be successfully built. Based on this, changing to a different distro does not have implications on our artifacts.

**Comment by djaglowski** [comment]
Closed via #7526
